src/main/java/pl/palonek/interview/course/service/CourseStatusService.java

```diff
@@ -0,0 +1,63 @@
+package pl.palonek.interview.course.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import pl.palonek.interview.course.model.CourseSession;
+import pl.palonek.interview.course.model.CourseSessionDTO;
+import pl.palonek.interview.course.model.SessionStatus;
+import pl.palonek.interview.course.repository.CourseRepository;
+
+import java.time.ZonedDateTime;
+
+
+@Service
+@RequiredArgsConstructor
+public class CourseStatusService {
+
+    private final CourseRepository courseRepository;
+
+    @Transactional
+    public void saveCourseStatus(CourseSessionDTO courseSessionDTO) {
+        String userId = courseSessionDTO.getUserId();
+        String courseId = courseSessionDTO.getCourseId();
+        ZonedDateTime timestamp = courseSessionDTO.getTimestamp();
```

Comment:
[minor] Unnecessary extraction of courseSessionDTO fields, initializeSession, updateSession, closeSession could take courseSessionDTO as argument.